### ISSUE TITLE
When root CA's DN name format is PrintableString, OpenSSL signed certificate with UTF8 format.

### ISSUE THREAD

**Issue by yuechen-chen** [openssl/openssl#18956]
OpenSSL should keep the DN name format the same as the root cert, shouldn’t change the encoding from PrintableString to utf8string. 
"
      Since name comparison assumes that attribute values encoded in
      different types (e.g., PrintableString and UTF8String) are assumed
      to represent different strings, any name components that appear in
      both the subject field and the issuer field SHOULD use the same
      encoding throughout the certification path.
”
From https://datatracker.ietf.org/doc/html/rfc4630

**Comment by mattcaswell** [comment]
Do you have some certs that demonstrate this issue?

**Comment by yuechen-chen** [comment]
Hi Matt, 

Thanks for replying!
Please find the the display of certificate server.crt below.

Note: 
- The root CA's common name is in PrintableString format. (see "122:d=5  hl=2 l=   7 prim:      PRINTABLESTRING   :tstroot")
- The certificate's common name in server.crt is in UTF8 format. (see "253:d=5  hl=2 l=   8 prim:      UTF8STRING        :testcase")
- When the certificate is created, parameter "string_mask" is unset in openssl.conf. 
- Command of create server certificate request: "openssl req -new -key server.key -out server.csr"
- Command of create server certificate "openssl x509 -CAcreateserial -req -days 365 -in server.csr -CA ca.crt -CAkey ca.key -out server.crt"


```
$ openssl asn1parse -inform PEM -in server.crt -i
    0:d=0  hl=4 l=1341 cons: SEQUENCE          
    4:d=1  hl=4 l= 805 cons:  SEQUENCE          
    8:d=2  hl=2 l=  20 prim:   INTEGER           :5CC6FFE350A9509F86D98906FCC9A54D4C725FF9
   30:d=2  hl=2 l=  13 cons:   SEQUENCE          
   32:d=3  hl=2 l=   9 prim:    OBJECT            :sha256WithRSAEncryption
   43:d=3  hl=2 l=   0 prim:    NULL              
   45:d=2  hl=2 l=  84 cons:   SEQUENCE          
   47:d=3  hl=2 l=  11 cons:    SET               
   49:d=4  hl=2 l=   9 cons:     SEQUENCE          
   51:d=5  hl=2 l=   3 prim:      OBJECT            :countryName
   56:d=5  hl=2 l=   2 prim:      PRINTABLESTRING   :XX
   60:d=3  hl=2 l=  21 cons:    SET               
   62:d=4  hl=2 l=  19 cons:     SEQUENCE          
   64:d=5  hl=2 l=   3 prim:      OBJECT            :localityName
   69:d=5  hl=2 l=  12 prim:      PRINTABLESTRING   :Default City
   83:d=3  hl=2 l=  28 cons:    SET               
   85:d=4  hl=2 l=  26 cons:     SEQUENCE          
   87:d=5  hl=2 l=   3 prim:      OBJECT            :organizationName
   92:d=5  hl=2 l=  19 prim:      PRINTABLESTRING   :Default Company Ltd
  113:d=3  hl=2 l=  16 cons:    SET               
  115:d=4  hl=2 l=  14 cons:     SEQUENCE          
  117:d=5  hl=2 l=   3 prim:      OBJECT            :commonName
  122:d=5  hl=2 l=   7 prim:      PRINTABLESTRING   :tstroot
  131:d=2  hl=2 l=  30 cons:   SEQUENCE          
  133:d=3  hl=2 l=  13 prim:    UTCTIME           :220809171112Z
  148:d=3  hl=2 l=  13 prim:    UTCTIME           :230809171112Z
  163:d=2  hl=2 l=  98 cons:   SEQUENCE          
  165:d=3  hl=2 l=  11 cons:    SET               
  167:d=4  hl=2 l=   9 cons:     SEQUENCE          
  169:d=5  hl=2 l=   3 prim:      OBJECT            :countryName
  174:d=5  hl=2 l=   2 prim:      PRINTABLESTRING   :US
  178:d=3  hl=2 l=  11 cons:    SET               
  180:d=4  hl=2 l=   9 cons:     SEQUENCE          
  182:d=5  hl=2 l=   3 prim:      OBJECT            :stateOrProvinceName
  187:d=5  hl=2 l=   2 prim:      UTF8STRING        :CA
  191:d=3  hl=2 l=  21 cons:    SET               
  193:d=4  hl=2 l=  19 cons:     SEQUENCE          
  195:d=5  hl=2 l=   3 prim:      OBJECT            :localityName
  200:d=5  hl=2 l=  12 prim:      UTF8STRING        :Default City
  214:d=3  hl=2 l=  28 cons:    SET               
  216:d=4  hl=2 l=  26 cons:     SEQUENCE          
  218:d=5  hl=2 l=   3 prim:      OBJECT            :organizationName
  223:d=5  hl=2 l=  19 prim:      UTF8STRING        :Default Company Ltd
  244:d=3  hl=2 l=  17 cons:    SET               
  246:d=4  hl=2 l=  15 cons:     SEQUENCE          
  248:d=5  hl=2 l=   3 prim:      OBJECT            :commonName
  253:d=5  hl=2 l=   8 prim:      UTF8STRING        :testcase
  263:d=2  hl=4 l= 546 cons:   SEQUENCE          
  267:d=3  hl=2 l=  13 cons:    SEQUENCE          
  269:d=4  hl=2 l=   9 prim:     OBJECT            :rsaEncryption
  280:d=4  hl=2 l=   0 prim:     NULL              
  282:d=3  hl=4 l= 527 prim:    BIT STRING        
  813:d=1  hl=2 l=  13 cons:  SEQUENCE          
  815:d=2  hl=2 l=   9 prim:   OBJECT            :sha256WithRSAEncryption
  826:d=2  hl=2 l=   0 prim:   NULL              
  828:d=1  hl=4 l= 513 prim:  BIT STRING 
```

Please let me know if you need any further information. Thanks so much for looking into this issue!

**Comment by mattcaswell** [comment]
Hmmm. I'm not sure why this would be considered a problem. The strings *are* different so why does it matter that one is PRINTABLESTRING and one is UTF8STRING?

**Comment by yuechen-chen** [comment]
In fact, it is a requirement from rfc4630 https://datatracker.ietf.org/doc/html/rfc4630. When the issuer's name is in PrintableString format, I think we should also use the same format for certificate's common name. Otherwise the same value will be encoded into different strings.

"
Since name comparison assumes that attribute values encoded in
different types (e.g., PrintableString and UTF8String) are assumed
to represent different strings, any name components that appear in
both the subject field and the issuer field SHOULD use the same
encoding throughout the certification path.
”

**Comment by bugspencor** [comment]
Researching a related (but not similar) issue, it turned out that RFC 4630 is obsoleted by RFC 5280 (https://www.rfc-editor.org/rfc/rfc5280), and the updated RFC doesn't include the quoted text anymore.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the end of 3.4 dev barring further input
